src/figuresClasses.ts

```diff
@@ -1,19 +1,83 @@
-export interface Figure {
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
+export interface Figure {
+  shape: Shape,
+  color: Color,
+  getArea: () => number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  public shape: Shape = 'triangle';
+
+  constructor(
+    public color: 'red' | 'green' | 'blue',
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('One of the figure\'s sides less or 0');
+    }
+
+    if (Math.max(a, b, c) >= (a + b + c) - Math.max(a, b, c)) {
+      throw new Error(
+        `throws an error: sides ${a}, ${b} and ${c} can't form a triangle`,
+      );
+    }
+  }
+
+  getArea(): number {
+    const { a, b, c } = this;
+    const s = (a + b + c) / 2; // semiperimeter
```

Comment:
Rename the variable to semiperimeter and remove the comment.

Reply:
akey